internal/auth0/resourceserver/resource_scope.go

```diff
@@ -0,0 +1,219 @@
+package resourceserver
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/auth0/go-auth0/management"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/auth0/terraform-provider-auth0/internal/config"
+)
+
+// NewScopeResource will return a new auth0_connection_client resource.
+func NewScopeResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"scope": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the scope.",
+			},
+			"resource_server_identifier": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Identifier of the resource server that the scope (permission) is associated with.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of the scope (permission).",
+			},
+		},
+		CreateContext: createResourceServerScope,
+		UpdateContext: updateResourceServerScope,
+		ReadContext:   readResourceServerScope,
+		DeleteContext: deleteResourceServerScope,
+		Importer: &schema.ResourceImporter{
+			StateContext: importResourceServerScope,
+		},
+		Description: "With this resource, you can manage scopes (permissions) associated with a resource server (API).",
+	}
+}
+
+func createResourceServerScope(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+
+	resourceServerID := data.Get("resource_server_identifier").(string)
+	scope := data.Get("scope").(string)
+	description := data.Get("description").(string)
+
+	currentScopes, err := api.ResourceServer.Read(resourceServerID)
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	scopes := append(currentScopes.GetScopes(), management.ResourceServerScope{
```

Comment:
What happens if the scope is already there when creating? As we're pontentially sending a duplicate here.

Reply:
Update: we have a dedicated ticket to tackle this here and in other potentially affected 1:1 resources.

Reply:
Update: we have a dedicated ticket to tackle this here and in other potentially affected 1:1 resources.